app/reports/emotion_index.rb

```diff
@@ -0,0 +1,78 @@
+class EmotionIndex < AdminReport
+  include Chartkick::Helper
+  def initialize(team, time_periods)
+    super(team)
+    @time_periods = time_periods
+  end
+
+  def generate
+    responses = receive_responses
+
+    return ['No emotion index present.', nil] if responses.empty?
+
+    positive_emotion_ids = positive_emotions(responses)
+    negative_emotion_ids = negative_emotions(responses)
+
+    positive_ratings_sum = responses.where(emotion_id: positive_emotion_ids).distinct.sum(:rating)
+    negative_ratings_sum = responses.where(emotion_id: negative_emotion_ids).distinct.sum(:rating)
+
+    total_responses = receive_total_responses
+
+    result = (positive_ratings_sum - negative_ratings_sum) / total_responses.to_f
```

Comment:
maybe we don't need this var total_responses, you can use receive_total_responses